app/black_spots/views.py

```diff
@@ -6,13 +9,47 @@
 
 from driver_auth.permissions import IsAdminOrReadOnly
 
+import uuid
+
 
 class BlackSpotViewSet(viewsets.ModelViewSet):
     """ViewSet for black spots"""
     queryset = BlackSpot.objects.all()
     serializer_class = BlackSpotSerializer
     filter_class = BlackSpotFilter
-    permission_classes = (IsAdminOrReadOnly,)
+    permission_classes = (IsAdminOrReadOnly, )
+
+    def list(self, request, *args, **kwargs):
+        # if tilekey if specified, use to get query
+        response = Response(None)
+        if ('tilekey' in request.query_params):
+            tile_token = request.query_params['tilekey']
+            redis_conn = get_redis_connection('default')
+            sql = redis_conn.get(tile_token)
+            if sql:
+                tilekey_queryset = BlackSpot.objects.raw(sql)
+                tilekey_serializer = BlackSpotSerializer(tilekey_queryset, many=True)
+                tilekey_serializer.data.insert(0, {'count': len(tilekey_serializer.data)})
+                response = Response(tilekey_serializer.data)
+        else:
+            response = super(BlackSpotViewSet, self).list(self, request, *args, **kwargs)
+        return response
+
+    def _generate_query_sql(self, request):
```

Comment:
I wonder if we should break this and the redis connection code out into a separate function or base class so that we don't repeat this logic; we'll probably need similar functionality for interventions as well.